pyccel/codegen/printing/ccode.py

```diff
@@ -2353,6 +2362,13 @@ def _print_PythonImag(self, expr):
     def _print_PythonConjugate(self, expr):
         return 'conj({})'.format(self._print(expr.internal_var))
 
+    def _print_ListAppend(self, expr):
+        target = expr.list_obj
+        class_type = target.class_type
+        dtype = self.get_c_type(class_type)
```

Comment:
Dtype is usually for the underlying datatype. Eg `int` in `list[int]` better to avoid this name for containers